modules/sentry/services/snuba.nix

```diff
@@ -0,0 +1,243 @@
+        default = "localhost";
+        description = ''
+          Host Redis is running on.
+        '';
+      };
+  
+      redisPort = mkOption {
+        type = types.int;
+        default = config.services.redis.port;
+        description = ''
+          Port Redis is running on.
+        '';
+      };
+  
+      redisDb = mkOption {
+        type = types.int;
+        default = 1;
+        description = ''
+          Redis database to use.
+        '';
+      };
+  
+      kafkaHost = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = ''
+          Host Kafka is running on.
+        '';
+      };
+  
+      kafkaPort = mkOption {
+        type = types.int;
+        default = config.services.apache-kafka.port;
+        description = ''
+          Port Kafka is running on.
+        '';
+      };
+  
+      clickhouseHost = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = ''
+          Host clickhouse is running on.
+        '';
+      };
+  
+      clickhouseClientPort = mkOption {
+        type = types.int;
+        default = 9000;
+        description = ''
+          Port clickhouse server is running on. Note that clickhouse
+          server listens on multiple ports, this port is the port
+          clickhouse client should use.
+        '';
+      };
+  
+      clickhouseHttpPort = mkOption {
+        type = types.int;
+        default = 8123;
+        description = ''
+          Port clickhouse HTTP server is running on. Note that
+          clickhouse server listens on multiple ports, this port is the
+          port for the HTTP server.
+        '';
+      };
+    };
+  };
+
+  config =
+    let
+      snubaSettingsPy = pkgs.writeText "settings.py" ''
+        import os
+        from snuba.settings_base import *  # NOQA
+      
+        env = os.environ.get
+  
+        def readPasswordFile(file):
+          with open(file, 'r') as fd:
+            fd.read()
+  
+        HOST = "${cfg.host}"
+        PORT = ${toString cfg.port}
+        
+        DEBUG = env("DEBUG", "0").lower() in ("1", "true")
+        
+        DEFAULT_BROKERS = "${cfg.kafkaHost}:${toString cfg.kafkaPort}".split(",")
+        
+        REDIS_HOST = "${cfg.redisHost}"
+        REDIS_PORT = ${toString cfg.redisPort}
+        REDIS_PASSWORD = ""
+        REDIS_DB = ${toString cfg.redisDb}
+        USE_REDIS_CLUSTER = False
+  
+        # Clickhouse Options
+        CLICKHOUSE_HOST = "${cfg.clickhouseHost}"
+        CLICKHOUSE_PORT = ${toString cfg.clickhouseClientPort}
+        CLICKHOUSE_HTTP_PORT = ${toString cfg.clickhouseHttpPort}
+        CLICKHOUSE_MAX_POOL_SIZE = 25
+        
+        # Dogstatsd Options
+        DOGSTATSD_HOST = None
+        DOGSTATSD_PORT = None
+      '';
+    in mkIf cfg.enable {
+      users.users.snuba = {
+        name = "snuba";
+        group = "snuba";
+        description = "Snuba user";
+        createHome = true;
+        uid = 103;
```

Comment:
could potentially conflict with
https://github.com/NixOS/nixpkgs/blob/14bc25038b1f2656bf30239ee7a04b0f863a7f3c/nixos/modules/misc/ids.nix#L142

Reply:
I've gone for dynamic uids and gids, we'll see if that works.

Reply:
Looks good :+1: